LinuxUtils.cpp

```diff
@@ -60,658 +60,587 @@ void fprintf_s_Linux(FILE* file, const char* format, ...)
 		return true;
 	}
+}
 
-	bool DoesFileExist(const std::string& path)
-	{
-	    const auto tempPath = fs::u8path(path);
-	    if (fs::exists(tempPath) && !is_directory(tempPath)) return true;
-	    return false;
-	}
+bool IsRegularFile(const std::string& path)
+{
+	return !fs::is_directory(fs::u8path(path));
+}
 
-	bool doesFolderExist(const std::string& path)
-	{
-	    const auto tempPath = fs::u8path(path);
-	    if (fs::exists(tempPath) && is_directory(tempPath)) return true;
-	    return false;
-	}
+bool IsDirectory(const std::string& path)
+{
+	return fs::is_directory(fs::u8path(path));
+}
 
-	void WriteToConsole(LogLevel level, const std::string& logMessage)
-	{
-		if(level != LogLevel::Debug) // Don't log debug messages to console.
-		{
-			std::cout << logMessage;
-		}
-	}
 
-	/*
-		Implemented all messages for error codes used in LinuxUtils.cpp.
-	*/
-	std::string GetLastErrorString()
+/*
+	Note: since the function signature did not allow for a return value, it clears the fileNames set when an
+	error occurs to make sure no operations are done on an incomplete list of files
+*/
+void GetAllFilesInFolderRecursive(const std::string& path, std::set<std::string>& filenames)
+{
+	for (auto& p: fs::recursive_directory_iterator(fs::u8path(path)))
 	{
-		using namespace std;
-		switch(errno)
+		if (!p.is_directory())
 		{
-			case EEXIST:
-				return string("Error code: EEXIST");
-			case EACCES:
-				return string("Error code: EACCES");
-			case ENOENT:
-				return string("Error code: ENOENT");
-			case ENOTDIR:
-				return string("Error code: ENOTDIR");
-			case EPERM:
-				return string("Error code: EPERM");
-			case EBUSY:
-				return string("Error code: EBUSY");
-			case ENOTEMPTY:
-				return string("Error code: ENOTEMPTY");
-			case EINVAL:
-				return string("Error code: EINVAL");
-			case EISDIR:
-				return string("Error code: EISDIR");
-			case EROFS:
-				return string("Error code: EROFS");	
-			case E2BIG:
-				return string("Error code: E2BIG");
-			case EILSEQ:
-				return string("Error code: EILSEQ");
-			default:
-				return string("Error code: unknown");
+			auto lastSlash = p.path().native().find_last_of("/");
+			auto tempDir = p.path().native().substr(0, lastSlash);
+			lastSlash = tempDir.find_last_of("/");
+			auto dirName = tempDir.substr(lastSlash + 1, tempDir.length());
+			auto returnName = "/" + dirName + "/" + p.path().filename().string();
+			filenames.insert(returnName);
 		}
 	}
+}
+
 
-	bool DeleteFile(const std::string &unresolvedFile)
+void WriteToConsole(LogLevel level, const std::string& logMessage)
+{
+	if (level != LogLevel::Debug) // Don't log debug messages to console.
 	{
-		fs::remove(fs::u8path(unresolvedFile));
-		return true;
+		std::cout << logMessage;
 	}
+}
 
-	bool deleteFolder(const std::string& unresolvedFolder)
-	{
-		fs::remove(fs::u8path(unresolvedFolder));
-		return true;
+/*
+	Implemented all messages for error codes used in LinuxUtils.cpp.
+*/
+std::string GetLastErrorString()
```

Comment:
Can maybe use [std::strerror()](https://en.cppreference.com/w/cpp/string/byte/strerror) instead. Though I thought I saw something else recently.

Reply:
strerror_s is the secure one, that one is depreciated. I've changed the function but haven't tested it yet. Need linux environment for that,